test/integration/test-amp-recaptcha-input.js

```diff
@@ -0,0 +1,210 @@
+            name="recaptcha-token"
+            data-sitekey="6LebBGoUAAAAAHbj1oeZMBU_rze_CutlbyzpH8VE"
+            data-action="recaptcha-example">
+          </amp-recaptcha-input>
+        </fieldset>
+
+        <div class="loading-spinner">
+          <div class="donut">
+          </div>
+        </div>
+
+        <div submit-success>
+          <template type="amp-mustache">
+            <div id="submit-success"></div>
+            <h1>You searched for: {{term}}</h1>
+            <p>message: {{message}}</p>
+            <p>recaptcha-token: {{recaptcha-token}}</p>
+          </template>
+        </div>
+
+        <div submit-error>
+          <template type="amp-mustache">
+            <div id="submit-error"></div>
+            <h1>Error! Please check the JS Console in your dev tools.</h1>
+            <p>{{error}}</p>
+          </template>
+        </div>
+
+      </form>
+    </div>
+    `,
+  css: `
+      form.amp-form-submit-success [submit-success] {
+        color: green;
+      }
+      form.amp-form-submit-error [submit-error] {
+        color: red;
+      }
+      form.amp-form-submit-success.hide-inputs > input {
+        display: none;
+      }
+
+      @keyframes donut-spin {
+        0% {
+          transform: rotate(0deg);
+        }
+        100% {
+          transform: rotate(360deg);
+        }
+      }
+
+      .donut {
+        display: inline-block;
+        border: 4px solid rgba(0, 0, 0, 0.1);
+        border-left-color: #7983ff;
+        border-radius: 50%;
+        width: 30px;
+        height: 30px;
+        animation: donut-spin 1.2s linear infinite;
+      }
+
+      .loading-spinner {
+        display: none;
+        text-align: center;
+        margin: 5px;
+      }
+
+      form.amp-form-submitting > .loading-spinner {
+        display: block;
+      }
+    `,
+  /* eslint-enable max-len */
+  extensions: ['amp-recaptcha-input', 'amp-form', 'amp-mustache'],
+  experiments: ['amp-recaptcha-input'],
+  timeout: 3000,
+}, env => {
+  let doc;
+
+  beforeEach(() => {
+    doc = env.win.document;
+
+    const browserController = new BrowserController(env.win);
+    return browserController.waitForElementLayout('amp-recaptcha-input');
+  });
+
+  it('should be able to create the bootstrap frame', function() {
+    this.timeout(7000);
+
+    return waitForBootstrapFrameToBeCreated(doc).then(frame => {
+      expect(frame.src.includes('recaptcha')).to.be.true;
+      expect(frame.getAttribute('data-amp-3p-sentinel'))
+          .to.be.equal('amp-recaptcha');
+      expect(frame.getAttribute('name')).to.be.equal(JSON.stringify({
+        'sitekey': '6LebBGoUAAAAAHbj1oeZMBU_rze_CutlbyzpH8VE',
+        'sentinel': 'amp-recaptcha',
+      }));
+    });
+  });
+
+  it('should load the 3p recaptcha frame', function() {
+    this.timeout(7000);
+    return waitForBootstrapFrameOnLoad(doc).then(frame => {
+      expect(frame).to.be.ok;
+    });
+  });
+
+  it('should create a hidden input, ' +
```

Comment:
it will be better if we can test the input kv pair is really sent over to server as body payload.

to do that, you can use request-bank.